<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>730</width>
    <height>700</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>730</width>
    <height>700</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="font">
   <font>
    <pointsize>16</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>3DViewer</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../build-3DViewer-Desktop_Qt_6_6_1_GCC_64bit-Debug/qmake_qmake_immediate.qrc">
    <normaloff>:/viewer.ico</normaloff>:/viewer.ico</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">color: rgb(10, 132, 255);
/*background-color: rgb(0, 0, 0);*/</string>
  </property>
  <property name="iconSize">
   <size>
    <width>128</width>
    <height>128</height>
   </size>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="maximumSize">
    <size>
     <width>8096</width>
     <height>4048</height>
    </size>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <layout class="QGridLayout" name="gridLayout_3" rowstretch="1,0">
    <property name="leftMargin">
     <number>20</number>
    </property>
    <property name="bottomMargin">
     <number>20</number>
    </property>
    <property name="horizontalSpacing">
     <number>6</number>
    </property>
    <property name="verticalSpacing">
     <number>20</number>
    </property>
    <item row="0" column="1">
     <layout class="QGridLayout" name="grid_scale">
      <item row="0" column="0" colspan="3">
       <widget class="QDoubleSpinBox" name="spin_box_scale">
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
        <property name="styleSheet">
         <string notr="true">QDoubleSpinBox { border: none; padding: 0px; }
QDoubleSpinBox:focus { border: none; outline: none; }</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="suffix">
         <string>%</string>
        </property>
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="minimum">
         <double>1.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1000.000000000000000</double>
        </property>
        <property name="value">
         <double>100.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSlider" name="slider_scale">
        <property name="styleSheet">
         <string notr="true">QSlider::groove:vertical {
    background-color: rgb(10, 132, 255); /* Изменение цвета полосы */
    width: 1px;
}
QSlider::handle:vertical {
    background-color: gray; /* Изменение цвета маркера*/
    height: 20px;
    margin: 0 -10px;
    border-radius: 10px;
}</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="singleStep">
         <number>1</number>
        </property>
        <property name="pageStep">
         <number>10</number>
        </property>
        <property name="value">
         <number>500</number>
        </property>
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <layout class="QGridLayout" name="grid_bottom" rowstretch="0,0,0,0" columnstretch="0,0,0,0,0" rowminimumheight="0,0,0,0">
      <property name="horizontalSpacing">
       <number>6</number>
      </property>
      <property name="verticalSpacing">
       <number>14</number>
      </property>
      <item row="0" column="0">
       <widget class="QDoubleSpinBox" name="spin_box_move_x">
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
        <property name="styleSheet">
         <string notr="true">QDoubleSpinBox { border: none; padding: 0px; }
QDoubleSpinBox:focus { border: none; outline: none; }</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="minimum">
         <double>-1.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QSlider" name="slider_rotate_y">
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="styleSheet">
         <string notr="true">QSlider::groove:horizontal {
    background-color: rgb(10, 132, 255); /* Изменение цвета полосы */
    height: 1px;
}
QSlider::handle:horizontal {
    background-color: gray; /* Изменение цвета маркера*/
    width: 20px;
    margin: -10px 0;
    border-radius: 10px;
}
</string>
        </property>
        <property name="minimum">
         <number>-180</number>
        </property>
        <property name="maximum">
         <number>180</number>
        </property>
        <property name="value">
         <number>0</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="label_y">
        <property name="font">
         <font>
          <pointsize>16</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Y</string>
        </property>
       </widget>
      </item>
      <item row="2" column="4">
       <widget class="QDoubleSpinBox" name="spin_box_rotate_z">
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
        <property name="styleSheet">
         <string notr="true">QDoubleSpinBox { border: none; padding: 0px; }
QDoubleSpinBox:focus { border: none; outline: none; }</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="suffix">
         <string>°</string>
        </property>
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="minimum">
         <double>-180.000000000000000</double>
        </property>
        <property name="maximum">
         <double>180.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="label_move">
        <property name="font">
         <font>
          <pointsize>16</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Перемещение</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSlider" name="slider_move_y">
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="styleSheet">
         <string notr="true">QSlider::groove:horizontal {
    background-color: rgb(10, 132, 255); /* Изменение цвета полосы */
    height: 1px;
}
QSlider::handle:horizontal {
    background-color: gray; /* Изменение цвета маркера*/
    width: 20px;
    margin: -10px 0;
    border-radius: 10px;
}
</string>
        </property>
        <property name="minimum">
         <number>-1000</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="singleStep">
         <number>10</number>
        </property>
        <property name="pageStep">
         <number>100</number>
        </property>
        <property name="value">
         <number>0</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QLabel" name="label_z">
        <property name="font">
         <font>
          <pointsize>16</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Z</string>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QDoubleSpinBox" name="spin_box_rotate_x">
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
        <property name="styleSheet">
         <string notr="true">QDoubleSpinBox { border: none; padding: 0px; }
QDoubleSpinBox:focus { border: none; outline: none; }</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="suffix">
         <string>°</string>
        </property>
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="minimum">
         <double>-180.000000000000000</double>
        </property>
        <property name="maximum">
         <double>180.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QDoubleSpinBox" name="spin_box_move_y">
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
        <property name="styleSheet">
         <string notr="true">QDoubleSpinBox { border: none; padding: 0px; }
QDoubleSpinBox:focus { border: none; outline: none; }</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="minimum">
         <double>-1.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QDoubleSpinBox" name="spin_box_rotate_y">
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
        <property name="styleSheet">
         <string notr="true">QDoubleSpinBox { border: none; padding: 0px; }
QDoubleSpinBox:focus { border: none; outline: none; }</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="suffix">
         <string>°</string>
        </property>
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="minimum">
         <double>-180.000000000000000</double>
        </property>
        <property name="maximum">
         <double>180.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QSlider" name="slider_rotate_z">
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="styleSheet">
         <string notr="true">QSlider::groove:horizontal {
    background-color: rgb(10, 132, 255); /* Изменение цвета полосы */
    height: 1px;
}
QSlider::handle:horizontal {
    background-color: gray; /* Изменение цвета маркера*/
    width: 20px;
    margin: -10px 0;
    border-radius: 10px;
}
</string>
        </property>
        <property name="minimum">
         <number>-180</number>
        </property>
        <property name="maximum">
         <number>180</number>
        </property>
        <property name="value">
         <number>0</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSlider" name="slider_move_x">
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="styleSheet">
         <string notr="true">QSlider::groove:horizontal {
    background-color: rgb(10, 132, 255); /* Изменение цвета полосы */
    height: 1px;
}
QSlider::handle:horizontal {
    background-color: gray; /* Изменение цвета маркера*/
    width: 20px;
    margin: -10px 0;
    border-radius: 10px;
}
</string>
        </property>
        <property name="minimum">
         <number>-1000</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="singleStep">
         <number>1</number>
        </property>
        <property name="pageStep">
         <number>100</number>
        </property>
        <property name="value">
         <number>0</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="3" column="3">
       <widget class="QLabel" name="label_rotate">
        <property name="font">
         <font>
          <pointsize>16</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Поворот</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSlider" name="slider_move_z">
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="styleSheet">
         <string notr="true">QSlider::groove:horizontal {
    background-color: rgb(10, 132, 255); /* Изменение цвета полосы */
    height: 1px;
}
QSlider::handle:horizontal {
    background-color: gray; /* Изменение цвета маркера*/
    width: 20px;
    margin: -10px 0;
    border-radius: 10px;
}
</string>
        </property>
        <property name="minimum">
         <number>-1000</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="singleStep">
         <number>10</number>
        </property>
        <property name="pageStep">
         <number>100</number>
        </property>
        <property name="value">
         <number>0</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="label_x">
        <property name="font">
         <font>
          <pointsize>16</pointsize>
         </font>
        </property>
        <property name="text">
         <string>X</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QDoubleSpinBox" name="spin_box_move_z">
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
        <property name="styleSheet">
         <string notr="true">QDoubleSpinBox { border: none; padding: 0px; }
QDoubleSpinBox:focus { border: none; outline: none; }</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="minimum">
         <double>-1.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QSlider" name="slider_rotate_x">
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="styleSheet">
         <string notr="true">QSlider::groove:horizontal {
    background-color: rgb(10, 132, 255); /* Изменение цвета полосы */
    height: 1px;
}
QSlider::handle:horizontal {
    background-color: gray; /* Изменение цвета маркера*/
    width: 20px;
    margin: -10px 0;
    border-radius: 10px;
}
</string>
        </property>
        <property name="minimum">
         <number>-180</number>
        </property>
        <property name="maximum">
         <number>180</number>
        </property>
        <property name="singleStep">
         <number>1</number>
        </property>
        <property name="value">
         <number>0</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout" stretch="1,18">
      <item>
       <widget class="QLabel" name="status_label">
        <property name="font">
         <font>
          <pointsize>16</pointsize>
         </font>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="s21::CustomGLWidget" name="open_gl_widget">
        <property name="styleSheet">
         <string notr="true"/>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>730</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Файл</string>
    </property>
    <widget class="QMenu" name="screenshot">
     <property name="title">
      <string>Cохранить скриншот</string>
     </property>
     <addaction name="separator"/>
     <addaction name="action_BMP"/>
     <addaction name="action_JPEG"/>
    </widget>
    <widget class="QMenu" name="gif">
     <property name="title">
      <string>Сохранить анимацию</string>
     </property>
     <addaction name="action10_FPS_5"/>
    </widget>
    <addaction name="choose_file"/>
    <addaction name="screenshot"/>
    <addaction name="gif"/>
    <addaction name="exit"/>
   </widget>
   <widget class="QMenu" name="help">
    <property name="title">
     <string>Справка</string>
    </property>
    <addaction name="about"/>
   </widget>
   <widget class="QMenu" name="view">
    <property name="title">
     <string>Вид</string>
    </property>
    <widget class="QMenu" name="projection">
     <property name="title">
      <string>Тип проекции</string>
     </property>
     <addaction name="projection_parallel"/>
     <addaction name="projection_central"/>
    </widget>
    <addaction name="projection"/>
    <addaction name="background_color"/>
    <addaction name="separator"/>
    <addaction name="settings"/>
    <addaction name="reset_settings"/>
    <addaction name="separator"/>
    <addaction name="theme"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="view"/>
   <addaction name="help"/>
  </widget>
  <action name="choose_file">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Открыть</string>
   </property>
   <property name="toolTip">
    <string>Открыть</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="about">
   <property name="text">
    <string>О программе</string>
   </property>
  </action>
  <action name="exit">
   <property name="text">
    <string>Выход</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="projection_parallel">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Параллельная</string>
   </property>
  </action>
  <action name="projection_central">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Центральная</string>
   </property>
  </action>
  <action name="background_color">
   <property name="text">
    <string>Цвет фона</string>
   </property>
  </action>
  <action name="settings">
   <property name="text">
    <string>Настройки</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+,</string>
   </property>
  </action>
  <action name="reset_settings">
   <property name="text">
    <string>Сбросить настройки</string>
   </property>
  </action>
  <action name="action10_FPS_5">
   <property name="text">
    <string>10 FPS, 5 s</string>
   </property>
  </action>
  <action name="action_JPEG">
   <property name="text">
    <string>в формате JPEG</string>
   </property>
  </action>
  <action name="action_BMP">
   <property name="text">
    <string>в формате BMP</string>
   </property>
  </action>
  <action name="theme">
   <property name="text">
    <string>Светлая тема</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>s21::CustomGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>widget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../build-3DViewer-Desktop_Qt_6_6_1_GCC_64bit-Debug/qmake_qmake_immediate.qrc"/>
 </resources>
 <connections/>
</ui>
